vm/src/stdlib/socket.rs

```diff
@@ -502,9 +502,7 @@ impl TryFromObject for Address {
             Err(vm.new_type_error("Address tuple should have only 2 values".to_string()))
         } else {
             Ok(Address {
-                host: PyStringRef::try_from_object(vm, tuple.elements[0].clone())?
-                    .value
-                    .to_string(),
+                host: vm.to_pystr(&tuple.elements[0])?,
```

Comment:
This actually convert the first argument in the tuple to string without validation.

Reply:
Ah, that's why the tests are failing.